### DISCUSSION TITLE
How to ssh/vnc into vm remotely

### DISCUSSION THREAD

**Discussion by samwyndham** [cirruslabs/tart#347] [Q&A] [answered]
Hi

Apologies if this is all very basic.

When I run `tart ip <some vm>` I currently get the _local_ ip address that can be used to access the vm from the host machine. Is it possible to instead ssh/vnc into the VM from a remote location not on the same network? How do I go about setting this up?

Thank you!

**Comment by edigaryev** [comment, marked as answer]
Normally you would use something like `tart run --net-bridged="Wi-Fi" <VM-NAME>` for that, but this functionality is currently blocked by the lack of entitlements in the Tart binary, see recent comments in https://github.com/cirruslabs/tart/issues/243.

**Reply by samwyndham** [reply]
That is a shame. I can access the VM with multiple SSH tunnels (see below) but I would prefer a way that doesn't need knowledge of the host machine.

```
ssh -L5900:localhost:5900 user@host
ssh -L5900:localhost:5900 user@vm
```

**Reply by mafreud** [reply]
@edigaryev Is this solved?
https://github.com/cirruslabs/tart/issues/243#issuecomment-1448902782

**Reply by fkorotkov** [reply]
It is partially solved. Bridged Networking works but `tart ip` will still return a local IP address. PR https://github.com/cirruslabs/tart/pull/446 will add an option to choose how Tart resolves IP via `--resolver` flag so `tart ip --resolve=arp` will return an IP address that you'll need for remote access.

**Reply by mafreud** [reply]
Ok. Thank you!
